egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,265 @@
+                      ", where egs.job.archive.scp generated randomly and examples are combined "
+                      " across all jobs as eg.archive.scp.")
+  parser.add_argument("--random-lang", type=str, action=nnet3_train_lib.StrToBoolAction, 
+                      help="If true, the lang-id in ranges.* selected"
+                      " w.r.t frequency distribution of remaining examples in each language,"
+                      " otherwise it is selected sequentially.",
+                      default=True, choices = ["false", "true"])
+  parser.add_argument("--max-archives", type=int, default=1000,
+                      help="max number of archives used to generate egs.*.scp");
+  parser.add_argument("--seed", type=int, default=1,
+                      help="Seed for random number generator")
+
+  parser.add_argument("--minibatch-size", type=int, default=512,
+                      help="The minibatch size used to generate scp files per job. "
+                           "It should be multiple of actual minibatch size.");
+  
+  parser.add_argument("--prefix", type=str, default="",
+                      help="Adds a prefix to the range files. This is used to distinguish between the train "
+                      "and diagnostic files.")
+
+  parser.add_argument("--lang2weight", type=str, 
+                      help="lang2weight file contains the weight per language to scale output posterior for that language.(format is: "
+                           "<lang-id> <weight>)");
+# now the positional arguments
+  parser.add_argument("num_langs", type=int,
+                      help="num of languages used in multilingual training setup.");
+  parser.add_argument("egs_scp_lists", type=str,
+                      help="list of egs.scp files per input language."
+                           "e.g. exp/lang1/egs/egs.scp exp/lang2/egs/egs.scp");
+
+  parser.add_argument("egs_dir",
+                      help="Name of egs directory e.g. exp/multilingual_a/egs");
+
+
+  print(' '.join(sys.argv))
+
+  args = parser.parse_args()
+ 
+  return args
+
+
+# Returns a random language number w.r.t 
+# amount of examples in each language.
+# It works based on sampling from a 
+# discrete distribution, where it returns i 
+# with prob(i) as (num_egs in lang(i)/ tot_egs).
+# tot_egs is sum of lang_len.
+def RandomLang(lang_len, tot_egs, random_selection):
+  assert(tot_egs > 0)
+  rand_int = random.randint(0, tot_egs - 1)
+  count = 0
+  for l in range(len(lang_len)):
+    if random_selection:
+      if rand_int > count and rand_int <= (count + lang_len[l]):
+        rand_lang = l
+        break
+      else:
+        count += lang_len[l]
+    else: 
+      if (lang_len[l] > 0):
+        rand_lang = l
+        break
+  assert(rand_lang >= 0 and rand_lang < len(lang_len))
+  return rand_lang
+
+# Read lang2weight file and return lang2weight array
+# where lang2weight[i] is weight for language i.
+def ReadLang2weight(lang2w_file):
```

Comment:
Consider using a dictionary where the keys are languages, rather than using a mapping from languages to indexes. This would help avoid bugs.